src/global/pidfile.cc

```diff
@@ -170,5 +173,9 @@ int pidfile_open(const md_config_t *conf)
     pfh.close();
     return -errno;
   }
+
+  pfh.pf_fd = fd;
```

Comment:
@dachary see https://github.com/ceph/ceph/pull/7463/files#r51405701 . and if we don't `remove()` in `close()`, it does not matter where we set the pfh member variables then.